lib/pg_eventstore/abstract_command.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require_relative 'event_serializer'
+require_relative 'pgresult_deserializer'
+require_relative 'queries'
+
+module PgEventstore
+  class AbstractCommand
+    attr_reader :queries
+    private :queries
+
+    # @param connection [PgEventstore::Connection]
+    # @param middlewares [Array<Object<#serialize, #deserialize>>]
+    # @param event_class_resolver [#call]
+    def initialize(connection, middlewares, event_class_resolver)
+      @queries = Queries.new(
+        connection,
+        EventSerializer.new(middlewares),
+        PgresultDeserializer.new(middlewares, event_class_resolver)
```

Comment:
Well, splitting on `read`/`write`  command won't work, because some commands doing both read and write in their implementation. But you are right, it is a bit confusing to have it here. Instead - it should accept `PgEventstore::Queries` instance. I'll improve this